lte/gateway/c/oai/lib/n11/SubscriberdbServiceClient.cpp

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <google/protobuf/util/time_util.h>
+#include <cassert>
+#include <grpcpp/impl/codegen/client_context.h>
+#include <grpcpp/impl/codegen/status.h>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+#include "lte/protos/subscriberauth.grpc.pb.h"
+#include "lte/protos/subscriberauth.pb.h"
+#include "SubscriberdbServiceClient.h"
+#include "ServiceRegistrySingleton.h"
+#include "3gpp_38.413.h"
+#include "amf_client_proto_msg_to_itti_msg.h"
+#include "log.h"
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::InsecureChannelCredentials;
+using grpc::Status;
+using magma::AsyncLocalResponse;
+using magma::ServiceRegistrySingleton;
+using magma::lte::M5GAuthenticationInformationAnswer;
+using magma::lte::M5GAuthenticationInformationRequest;
+
+extern task_zmq_ctx_t grpc_service_task_zmq_ctx;
+
+static void handle_subs_authentication_info_ans(
+    grpc::Status status,
+    magma::lte::M5GAuthenticationInformationAnswer response,
+    const std::string& imsi, uint8_t imsi_length, amf_ue_ngap_id_t ue_id) {
+  MessageDef* message_p;
+
+  if (!status.ok()) {
+    std::cout << "get_subs_auth_info fails with code " << status.error_code()
+              << ", msg: " << status.error_message() << std::endl;
+
+    return;
+  }
+
+  message_p =
+      itti_alloc_new_message(TASK_GRPC_SERVICE, AMF_APP_SUBS_AUTH_INFO_RESP);
+  if (!message_p) {
+    // OAILOG_ERROR(
+    //    LOG_UTIL, "Message IP Allocation Response allocation failed\n");
+    return;
+  }
```

Comment:
this can be removed as `itti_alloc_new_message` will assert rather than return null.

Reply:
Comments addressed.